students/toddschultz/session08/circle.py

```diff
@@ -0,0 +1,74 @@
+from math import pi
+
+class Circle:
+
+    ''' Making Circles with codes. '''
+    def __init__(self, radius):
+        self.radius = radius 
+        self._diameter = radius * 2
+        self._area = pi * radius**2
+
+
+    @property
+    def diameter(self):
+        return self._diameter
+
+    @diameter.setter 
+    def diameter(self, diameter):
+        if diameter < 1:
+            raise ValueError
+        else:
+            self._diameter = diameter
+            self.radius = diameter / 2
+
+    @property
+    def area(self):
+        return self._area
+
+    @classmethod
+    def from_diameter(cls, diameter):
+        if diameter <= 0:
+            raise ValueError
+        self = cls(diameter / 2)
+        self = self
+        self.diameter = diameter
+        return self
```

Comment:
this can be simplified:
```
@classmethod
    def from_diameter(cls, diameter):
        if diameter <= 0:
            raise ValueError
        return cls(diameter / 2.0)
```